### PR TITLE
Add org.diasurgical.DevilutionX

### PR DESCRIPTION
**Pull request by tsunamistate**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: org.diasurgical.DevilutionX.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      sources:
+        - type: archive
+          url: https://github.com/diasurgical/devilutionX/releases/download/1.5.0/devilutionx-src-fully-vendored.tar.xz
+          sha256: 18aee4324c71198a26f15fe2b342552411c2f62039b64cf9b2fa3e249f1bc4f9
+        # CJK fonts
+        - type: file
+          filename: fonts.mpq
+          url: https://github.com/diasurgical/devilutionx-assets/releases/download/v3/fonts.mpq
+          sha256: 6b62e03c42ae4427055e0f292a1beafb5e840c397adaf1f641b909be37b8653e
+          size: 58488019
```
Comment (hfiguiere) [inline org.diasurgical.DevilutionX.yml:32]:
```suggestion
```

File: org.diasurgical.DevilutionX.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        # Diablo shareware demo
+        - type: file
+          filename: spawn.mpq
+          url: https://github.com/diasurgical/devilutionx-assets/releases/download/v3/spawn.mpq
+          sha256: 64427cd7c1ba904eaa2e0031c16a6b136d0ecef9abc888c5ff8344b459356e38
+          size: 25448219
```
Comment (hfiguiere) [inline org.diasurgical.DevilutionX.yml:38]:
```suggestion
```

File: org.diasurgical.DevilutionX.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      post-install:
+        - "sed -i 's/Icon=devilutionx-hellfire/Icon=org.diasurgical.DevilutionX-hellfire/g' ${FLATPAK_DEST}/share/applications/devilutionx-hellfire.desktop"
```
Comment (hfiguiere) [inline org.diasurgical.DevilutionX.yml:40]:
```suggestion
```

1. `rename-icon` takes care of it
2. using `destkop-file-edit` is always prefered.

Comment (hfiguiere) [inline org.diasurgical.DevilutionX.yml:40]: let's try this again

```suggestion
        - "desktop-file-edit --set-icon=org.diasurgical.DevilutionX-hellfire ${FLATPAK_DEST}/share/applications/devilutionx-hellfire.desktop"
```

using `destkop-file-edit` is always prefered.

File: org.diasurgical.DevilutionX.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        - "mv ${FLATPAK_DEST}/share/applications/devilutionx-hellfire.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}-hellfire.desktop"
+        - "mv ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/devilutionx-hellfire.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/${FLATPAK_ID}-hellfire.png"
+        - "mv fonts.mpq ${FLATPAK_DEST}/share/diasurgical/devilutionx"
+        - "mv spawn.mpq ${FLATPAK_DEST}/share/diasurgical/devilutionx"
```
Comment (hfiguiere) [inline org.diasurgical.DevilutionX.yml:44]:
```suggestion
        - "install -Dm644 fonts.mpq -t ${FLATPAK_DEST}/share/diasurgical/devilutionx"
        - "install -Dm644  spawn.mpq -t ${FLATPAK_DEST}/share/diasurgical/devilutionx"
```

File: org.diasurgical.DevilutionX.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
+app-id: org.diasurgical.DevilutionX
```
Comment (hfiguiere) [inline org.diasurgical.DevilutionX.yml:1]: the domain `diasurgical.org` seem to ne parked. does it belong to the developers?

**Comment by hfiguiere** [comment]
bot, build org.diasurgical.DevilutionX

**Comment by tsunamistate** [comment]
Looked at the build fails

The changes required were already done here: https://github.com/diasurgical/devilutionX/pull/6245
But they were done AFTER 1.5.0 release, so they're not in the source tarball
Looks like I'll have to wait for 1.5.1 release, which will be soon anyway

File: org.diasurgical.DevilutionX.yml (hunk 5 above)
Comment (AJenbo) [reply org.diasurgical.DevilutionX.yml:1]: yes

Comment (tsunamistate) [reply org.diasurgical.DevilutionX.yml:1]: I'll change the domain to `io.github.diasurgical.DevilutionX`

Comment (AJenbo) [reply org.diasurgical.DevilutionX.yml:1]: why? I actually own the diasurgical.org domain, that's what we use to validate for Google Play.
Use what we use elsewhere: `org.diasurgical.devilutionx`

Comment (AJenbo) [reply org.diasurgical.DevilutionX.yml:1]: Let me know if you need to verify ownership or something.

Comment (tsunamistate) [reply org.diasurgical.DevilutionX.yml:1]: Oh, that's great then!

Comment (AJenbo) [reply org.diasurgical.DevilutionX.yml:1]:
```suggestion
app-id: org.diasurgical.devilutionx
```

Comment (tsunamistate) [reply org.diasurgical.DevilutionX.yml:1]: If you look at other app IDs, unless the app name is stylized as lower-case, it stays capitalized

**Comment by hfiguiere** [comment]
you can still add a patch

**Comment by tsunamistate** [comment]
* Applied suggestions from review
* Added a patch to fix metainfo
* Uploaded screenshots to reference in metainfo later, currently it's using Discord attachments

File: fix-metainfo-1.5.0.patch
Comment (hfiguiere) [inline fix-metainfo-1.5.0.patch:1]: this patch could be smaller.

File: screens/screen1.png
Comment (hfiguiere) [inline screens/screen1.png:1]: why are these screenshots here if they are not referenced in the appstream file.

File: fix-metainfo-1.5.0.patch
Comment (tsunamistate) [reply fix-metainfo-1.5.0.patch:1]: Spacing on previous file was wrong

**Comment by hfiguiere** [comment]
bot, build org.diasurgical.DevilutionX

**Comment by AJenbo** [comment]
can i also have permissions?

**Comment by barthalion** [comment]
Invite sent!

**Pull request closed without merging**